pkg/didcomm/crypto/jwe/authcrypt/authcrypt_test.go

```diff
@@ -8,56 +8,121 @@ package authcrypt
 
 import (
 	"bytes"
+	"crypto/ed25519"
+	"crypto/rand"
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/require"
-	chacha "golang.org/x/crypto/chacha20poly1305"
-	"golang.org/x/crypto/nacl/box"
 
 	"github.com/hyperledger/aries-framework-go/pkg/internal/cryptoutil"
 	mockwallet "github.com/hyperledger/aries-framework-go/pkg/internal/mock/wallet"
 )
 
 func TestEncrypt(t *testing.T) {
 	var err error
-	var ecKeyPub *[chacha.KeySize]byte
-	var ecKeyPriv *[chacha.KeySize]byte
-	// create temporary keys for testing
-	ecKeyPub, ecKeyPriv, err = box.GenerateKey(randReader)
+	// create temporary signing keys for tests
+	sigPubKey, sigPrivKey, err := ed25519.GenerateKey(rand.Reader)
 	require.NoError(t, err)
-	senderKp := cryptoutil.KeyPair{Priv: ecKeyPriv[:], Pub: ecKeyPub[:]}
-	t.Logf("sender key pub: %v", base64.RawURLEncoding.EncodeToString(senderKp.Pub))
-	t.Logf("sender key priv: %v", base64.RawURLEncoding.EncodeToString(senderKp.Priv))
+	// convert signing keys to encryption keys
+	encPubKey, err := cryptoutil.PublicEd25519toCurve25519(sigPubKey)
+	require.NoError(t, err)
+	encPrivKey, err := cryptoutil.SecretEd25519toCurve25519(sigPrivKey)
+	require.NoError(t, err)
+
+	require.NoError(t, err)
+
+	sender := &cryptoutil.KeyPairCombo{
+		SigKeyPair: &cryptoutil.SigKeyPair{
+			KeyPair: cryptoutil.KeyPair{Pub: sigPubKey, Priv: sigPrivKey},
+			Alg:     cryptoutil.EdDSA,
+		},
+		EncKeyPair: &cryptoutil.EncKeyPair{
+			KeyPair: cryptoutil.KeyPair{Pub: encPubKey, Priv: encPrivKey},
+			Alg:     cryptoutil.Curve25519,
+		},
+	}
+	t.Logf("sender Signature key pub: %v", base64.RawURLEncoding.EncodeToString(sender.SigKeyPair.Pub))
+	t.Logf("sender Signature key priv: %v", base64.RawURLEncoding.EncodeToString(sender.SigKeyPair.Priv))
+	t.Logf("sender Encryption key pub: %v", base64.RawURLEncoding.EncodeToString(sender.EncKeyPair.Pub))
+	t.Logf("sender Encryption key priv: %v", base64.RawURLEncoding.EncodeToString(sender.EncKeyPair.Priv))
 
-	ecKeyPub, ecKeyPriv, err = box.GenerateKey(randReader)
+	sigPubKey, sigPrivKey, err = ed25519.GenerateKey(rand.Reader)
 	require.NoError(t, err)
-	recipient1Kp := cryptoutil.KeyPair{Priv: ecKeyPriv[:], Pub: ecKeyPub[:]}
-	t.Logf("recipient1Kp pub: %v", base64.RawURLEncoding.EncodeToString(recipient1Kp.Pub))
-	t.Logf("recipient1Kp priv: %v", base64.RawURLEncoding.EncodeToString(recipient1Kp.Priv))
+	// convert signing keys to encryption keys
+	encPubKey, err = cryptoutil.PublicEd25519toCurve25519(sigPubKey)
```

Comment:
Maybe pull this `KeyPairCombo` construction logic out to a helper function in this file